crypto/ec/ecx_meth.c

```diff
@@ -31,6 +31,11 @@
 #define KEYLENID(id)    (IS25519(id) ? X25519_KEYLEN \
                                      : ((id) == EVP_PKEY_X448 ? X448_KEYLEN \
                                                               : ED448_KEYLEN))
+#define KEYNID2TYPE(id) \
+    (IS25519(id) ?  ECX_KEY_TYPE_X25519 \
+                 : ((id) == EVP_PKEY_X448 ? ECX_KEY_TYPE_X448 \
+                                          : ((id) == EVP_PKEY_ED25519 ? ECX_KEY_TYPE_ED25519 \
```

Comment:
Isn't the first `IS25519` going to catch both x25519 and ed25519?  I'm not sure how this last `EVP_PKEY_ED25519` check is going to do anything.